### DISCUSSION TITLE
Building ResNet50 with FINN-HLS

### DISCUSSION THREAD

**Discussion by dtchuink** [Xilinx/finn#544] [Q&A]
Hi,

 I am trying to build ResNet50 with FINN-HLS. However, I have an issue with the first convolution. It has the following parameters:
```
KERNEL_DIM 7 // must be odd
IFMDim 224
CHANNEL_IN 3
CHANNEL_OUT 64
PADDING 3
STRIKE 2
```

When I am trying to synthesize with a configuration (PE, SIMD) other than 1, it does not work (Port ' ' has not fanin or fanout ERROR)

1. I figured that after adding the padding, the IFMDim = 229, which is a prime number. Can it justify the error? 
2. Should I call a **Stream Padding** function before the convolution and adjust the dimension?

Any input is welcome.

Best,
Danielle

**Comment by nhma20** [comment]
There is a ResNet50 example in the finn-examples repository (https://github.com/Xilinx/finn-examples/tree/main/build/resnet50). Maybe you can draw some inspiration from there.
